modules/porous_flow/src/materials/PorousFlowPorosityLinear.C

```diff
@@ -0,0 +1,195 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowPorosityLinear.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowPorosityLinear);
+
+InputParameters
+PorousFlowPorosityLinear::validParams()
+{
+  InputParameters params = PorousFlowPorosityBase::validParams();
+  params.addParam<bool>("strain_at_nearest_qp",
+                        false,
+                        "When calculating nodal porosity that depends on strain, use the strain at "
+                        "the nearest quadpoint.  This adds a small extra computational burden, and "
+                        "is not necessary for simulations involving only linear lagrange elements. "
+                        " If you set this to true, you will also want to set the same parameter to "
+                        "true for related Kernels and Materials");
+  params.addRequiredCoupledVar(
+      "porosity_ref",
+      "The porosity at reference effective porepressure, temperature and volumetric strain.  This "
+      "must be a real number or a constant monomial variable (not a linear lagrange or other type "
+      "of variable)");
+  params.addCoupledVar(
+      "P_ref",
+      0.0,
+      "The reference effective porepressure.  This should usually be a linear-lagrange variable");
+  params.addCoupledVar(
+      "T_ref",
+      0.0,
+      "The reference temperature.  This should usually be a linear-lagrange variable");
+  params.addCoupledVar("epv_ref",
+                       0.0,
+                       "The reference volumetric strain.  This must be a real number or a constant "
+                       "monomial variable (not a linear lagrange or other type of variable)");
+  params.addParam<Real>("P_coeff", 0.0, "Effective porepressure coefficient");
+  params.addParam<Real>("T_coeff", 0.0, "Temperature coefficient");
+  params.addParam<Real>("epv_coeff", 0.0, "Volumetric-strain coefficient");
+  params.addParam<Real>("porosity_min",
```

Comment:
Should this be range checked to be 0 < phi_min < 1?

Reply:
Certainly 0<=phi_min.  But not the upper bound, as I want to use this for lower-dimensional problems, where porosity can be greater than 1, because the porosity you enter into the input file is actually aperture*3Dporosity.